### PR TITLE
update the set of supported IntelliJ versions

### PR DESCRIPTION
**Pull request by devoncarew**
- update the set of supported IntelliJ versions

This drops two older IntelliJ versions (2020.1 and 2020.2).

### CODE REVIEW

**Comment by helin24** [comment]
Is this for v57? I believe @stevemessick is testing that version now already. 

I'm not sure if we normally change `edit.dart` with these versioning changes, but you may be able to remove old code substitutions as well.

**Comment by devoncarew** [comment]
> Is this for v57? I believe @stevemessick is testing that version now already.

It could be but it's not necessary - we just don't need to publish for these versions anymore.

> I'm not sure if we normally change `edit.dart` with these versioning changes, but you may be able to remove old code substitutions as well.

That's a good idea - I'll take a look at what's no longer used.

**Comment by devoncarew** [comment]
edit.dart is now updated; there were a fair number of changes there so its worth looking over

**Review by helin24 — APPROVED** [review]
LGTM, maybe Steve will think of other items. We should wait for him to merge since he may want to keep this out of master until v57 is released.

File: .github/workflows/presubmit.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        version: [4.1, 4.2, AF.3.1, 2020.4, 2021.1]
+        version: [AF.3.1, 2020.4, 2021.1]
```
Comment (stevemessick) [inline .github/workflows/presubmit.yaml:21]: Android Studio 4.2 is the current stable release, so we need to support it.

File: product-matrix.json [hunk 2]
```diff
@@ -1,32 +1,5 @@
...
-      "ideaVersion": "201.6565218",
-      "dartPluginVersion": "201.7223.99",
-      "sinceBuild": "201.7223",
-      "untilBuild": "201.*",
-      "filesToSkip": [
-        "src/io/flutter/utils/AndroidLocationProvider.java",
-        "flutter-studio/src/io/flutter/actions/FlutterNewProjectAction.java"
-      ]
-    },
-    {
```
Comment (stevemessick) [inline product-matrix.json:18]: We can't delete this yet.

**Review by stevemessick — CHANGES_REQUESTED** [review]

File: product-matrix.json (hunk 2 above)
Comment (devoncarew) [reply product-matrix.json:18]: This should only be used by 2020.1.1 / AS 4.1?

File: .github/workflows/presubmit.yaml (hunk 1 above)
Comment (devoncarew) [reply .github/workflows/presubmit.yaml:21]: Ah, this was inadvertent; I'll re-add.

File: product-matrix.json (hunk 2 above)
Comment (devoncarew) [reply product-matrix.json:18]: Ah, you mean the section below.

File: tool/plugin/lib/edit.dart [hunk 3]
```diff
@@ -175,19 +135,9 @@ class EditAndroidModuleLibraryManager extends EditCommand {
...
-        source = source.replaceAll(
-            "androidProject.init42", "androidProject.initPre41");
-        source = source.replaceAll("PluginManagerCore.getLoadedPlugins(), null",
-            "PluginManagerCore.getLoadedPlugins(), false");
-      }
       if (spec.version.startsWith('4.2')) {
-        source = source.replaceAll(
-            "androidProject.init42", "androidProject.init41");
+        source =
+            source.replaceAll("androidProject.init42", "androidProject.init41");
```
Comment (devoncarew) [inline tool/plugin/lib/edit.dart:140]: We might be able to update this as well?

Comment (stevemessick) [reply tool/plugin/lib/edit.dart:140]: Best to leave that alone for now. I'm hoping to delete it sometime soon.

**Review by stevemessick — APPROVED** [review]
I'll publish M57 with support for 4.1. No need to merge this until June.

**Comment by devoncarew** [comment]
Landing this now -

**Pull request merged**